src/main/java/com/mapzen/valhalla/Route.kt

```diff
@@ -1,20 +1,48 @@
 package com.mapzen.valhalla
 
 import android.location.Location
-import com.f2prateek.ln.Ln
 import com.mapzen.helpers.GeometryHelper.getBearing
 import org.json.JSONArray
 import org.json.JSONObject
 import java.lang.Math.toRadians
-import java.util.ArrayList
-import java.util.HashSet
+import java.util.*
 
 public open class Route {
+
+    private val TAG = Route::class.java.simpleName
+
     public val SNAP_PROVIDER: String = "snap"
+    /**
+     * Distance in meters
+     */
+    public val CLOSE_TO_DESTINATION_THRESHOLD: Int = 20
+    /**
+     * Distance in meters
+     */
+    public val CLOSE_TO_NEXT_LEG_THRESHOLD: Int = 5
+    /**
+     * Distance in meters
+     */
     public val LOST_THRESHOLD: Int = 50
-    public val CLOCKWISE: Int = 90
-    public val COUNTERCLOCKWISE: Int = -90
+    /**
+     * Distance in meters
+     *
+     * When at the beginning of a route, use a larger lost threshold in case there are
+     * parking lots or other obstacles user must navigate to get to path
+     */
+    public val BEGINNING_ROUTE_LOST_THRESHOLD: Int = 1000
```

Comment:
Instead of using an arbitrary value here, what if we made this a variable equal to the distance from the user's starting location to the first node plus the standard lost threshold? For example:

```kotlin
public var beginningRouteLostThresholdInMeters: Int? = null

...

public open fun snapToRoute(currentLocation: Location): Location? {
    if (beginningRouteLostThresholdInMeters == null {
        beginningRouteLostThresholdInMeters = currentLocation.distanceTo(poly[0].getLocation) + LOST_THRESHOLD
    }

    ...
}
```

Currently ongoing reroutes would still be triggered if the user is over 1K meters from the closest routeable area. Also this would guard against the scenario when the user is actually quite close to the start of the route (~100m) but then starts traveling in the opposite direction in which case a reroute would not be triggered for another 900m.